app/src/main/java/com/beemdevelopment/aegis/importers/AuthenticatorProImporter.java

```diff
@@ -0,0 +1,280 @@
+        }
+        
+        private Result convertSql() throws OtpInfoException {
+            assert this.entries != null;
+            Result ret = new Result();
+            for (SqlEntry entry : this.entries) {
+                try {
+                    ret.addEntry(convertEntry(entry));
+                } catch (DatabaseImporterEntryException e) {
+                    ret.addError(e);
+                }
+            }
+            
+            return ret;
+        }
+        
+        private static VaultEntry convertEntry(JSONObject authenticator) throws JSONException, EncodingException, OtpInfoException, DatabaseImporterEntryException {
+            int type = authenticator.getInt("Type");
+            String issuer = authenticator.getString("Issuer");
+            Object nullableUsername = authenticator.get("Username");
+            String username = nullableUsername == JSONObject.NULL ? "" : nullableUsername.toString();
+            byte[] secret = Base32.decode(authenticator.getString("Secret"));
+            Algorithm algo = Algorithm.values()[authenticator.getInt("Algorithm")];
+            int digits = authenticator.getInt("Digits");
+            int period = authenticator.getInt("Period");
+            int counter = authenticator.getInt("Counter");
+
+            return fromAny(type, issuer, username, secret, algo, digits, period, counter, authenticator);
+        }
+        
+        private static VaultEntry convertEntry(SqlEntry entry) throws DatabaseImporterEntryException, OtpInfoException {
+            return entry.convert();
+        }
+    }
+    
+    private static class SqlEntry extends SqlImporterHelper.Entry {
+        private final int type;
+        private final String issuer;
+        private final String username;
+        private final byte[] secret;
+        private final Algorithm algo;
+        private final int digits;
+        private final int period;
+        private final int counter;
+        public SqlEntry(Cursor cursor) {
+            super(cursor);
+            this.type = SqlImporterHelper.getInt(cursor, "type");
+            this.issuer = SqlImporterHelper.getString(cursor, "issuer");
+            this.username = SqlImporterHelper.getString(cursor, "username");
+            String secret = SqlImporterHelper.getString(cursor, "secret");
+            try {
+                this.secret = Base32.decode(secret);
+            } catch (EncodingException e) {
+                throw new SQLiteException(secret); // Rethrown upstream as DatabaseImporterException
+            }
+            this.algo = Algorithm.values()[SqlImporterHelper.getInt(cursor, "algorithm")];
+            this.digits = SqlImporterHelper.getInt(cursor, "digits");
+            this.period = SqlImporterHelper.getInt(cursor, "period");
+            this.counter = SqlImporterHelper.getInt(cursor, "counter");
+        }
+
+        @SuppressLint("DefaultLocale")
+        @NotNull
+        @Override
+        public String toString() {
```

Comment:
What is this used for?

Reply:
It appears to be part of an old prototype that i didn't notice before committing. Will be removed.

Reply:
Ah  actually, it is used in https://github.com/beemdevelopment/Aegis/blob/8a22acc9959f031f2c672e53ad8234e2595d72b1/app/src/main/java/com/beemdevelopment/aegis/importers/AuthenticatorProImporter.java#L118
An idea i looked at other importers to see what they did regarding that